API_DOCUMENTATION.md

```diff
@@ -0,0 +1,414 @@
+
+```javascript
+NucleoObject(NucleoObject);
+```
+
+**Usage**
+
+```javascript
+import {
+  NucleoString,
+  NucleoNumber,
+  NucleoBoolean,
+  NucleoObject,
+  NucleoList
+} from 'nucleojs'
+
+const fullNameContract = new NucleoObject({
+  name: 'fullName',
+  fields:  {
+    firstName: NucleoString,
+    lastName: NucleoString
+  }
+});
+
+const userContract = new NucleoObject({
+  name: 'user', // don't need to be the same name as the variable, but need to be unique
+  fields: {
+    name: fullNameContract,
+    age: NucleoNumber,
+    disabled: NucleoBoolean,
+  }
+});
+
+const usersContract = new NucleoList(userContract);
+
+const itemsContract = new NucleoList(NucleoString);
+
+const contracts = {
+  users: usersContract,
+  items: itemsContract
+};
+```
+
+## Creating NucleoString
+
+```javascript
+NucleoString: String;
+```
+
+**Usage**
+
+```javascript
+const name = 'Jhon';
+```
+
+## Creating NucleoNumber
+
+```javascript
+NucleoNumber: Number;
+```
+
+**Usage**
+
+```javascript
+const age = 27;
+```
+
+## Creating NucleoBoolean
+
+```javascript
+NucleoBoolean: Boolean;
+```
+
+**Usage**
+
+```javascript
+const disabled = false;
+```
+
+## Creating the store
+
+```javascript
+import { createStore } from 'nucleojs';
+import * as contracts from './contracts';
+
+const store = createStore(contracts); // send contracts to create the store
+const { dispatch, update, getStore, subscribe } = store; // these 4 functions are returned from store creation
+```
+
+## Dispatching and updating the store
+
+Nucleo provides two methods of saving data, used for different approaches.
+
+**dispatch:** works for saving data according to the full contract, used to save the very first contract state in the store or to update the whole contract in the store;
+
+**update:** works for updating parts of data, it performs a index search in the object and save it. `update` will fail if you try to first save a contract to the store using it.
+
+---
+
+Dispatch function, considering user contract above:
+
+```javascript
+
+let user = dispatch('user')({ name: { firstName: 'John', lastName: 'Nor' } });
+// it'll fail because it's missing age field
+
+user = dispatch('user')({ name: { firstName: 'John', lastName: 'Nor' }, age: 27 });
+// it'll save the data to store properly
+
+console.log(user);
+/*
+{
+  status: 'OK',
+  errors: [],
+  data: {
+    name: {
+      firstName: 'John',
+      lastName: 'Nor'
+    },
+    age: 27
+  },
+}
+*/
+```
+
+Update function, considering user contract above:
+
+```javascript
+const user = update('user')({ name: { firstName: 'Robert' }});
+// it'll update only the user first name and only if this item has been already created in the store before
+
+console.log(user);
+/*
+{
+  status: 'OK',
+  errors: [],
+  data: {
+    name: {
+      firstName: 'Robert',
+      lastName: 'Nor'
+    },
+    age: 27
+  },
+}
+*/
+```
+
+## Get contracts in store
+
+The `getStore` function returns all contracts that are in the store.
```

Comment:
Let's merge that, but I'm working on some architecture to do not have it, to get data in a way that this data is safe and immutable for real. We can chat further about it